src/components/qas/questions/qaCalculator/qaCalculatorsData.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { QACalculatorIcon, QACalculatorIconUnit } from './qaCalculatorIcons';
+import FindImpactCrater from './equations/FindImpactCrater';
+import FindDiameter from './equations/FindDiameter';
+import FindKineticEnergy from './equations/FindKineticEnergy';
+import {
+  getCalculatedMeasurementsForDistance,
+  calculateDiameter,
+  calculateKineticEnergy,
+  calculateCraterDiameter,
+} from './qaCalculatorUtilities';
+import { col50 } from './qaCalculator.module.scss';
+import Unit from '../../../charts/shared/unit';
+
```

Comment:
You're right.  This is not what I had in mind, but it certainly does a great job of consolidating the work and organizing information unique to each calculator in a manner that's both fairly readable, and fairly easy to update info in the future.  I think it's perhaps debatable whether this approach allows for greater flexibility.  And I think the Distance Calculator is a good example of how this: you had to create a condition outside of this system of calc definitions, that uses a hardcoded string value to conditionally render a pretty sizable chunk of mark up, a layout/style, I might add, I may already be anticipating using in the Impact Calculator.  Additionally, with this approach, if you want/need to change the name of a prop, you'll not only have to update it wherever that prop is used but also across this file as well.  I'm not suggesting this is a bad approach, I just want to recognize that it's not without its complications.

I think what I would have done is used the framework of react components as the organizational system.  So, rather than have a js object that represents the info unique to a calc, you would have had a React component for each calc that contains all the unique markup, styles, logic etc.  Any styles, logic, or markup not unique to the calc can be passed into it as props (in the traditional way or by way of a Higher Order Component), or live in a parent component itself (as you have done).  That would mean things like the active states, answerable, all the event handlers, some styles, some markup, etc.  And anything unique to a calc, like its form fields, markup, the math, how answers are presented, lives in its own component.  And in some cases, as you have done, even some of those unique things can be abstracted into their own reusable components (if sufficiently similar).  

I'm not at all suggesting you revisit this work to affect "how I would have done it," but I did want to share my perspective, and invite you to discuss it further if I have not well described my competing approach.

Reply:
I would love to see your perspective on this. I was very much stuck, in a way, that I was very uncomfortable with "how to do it the react way". The approach was not clear to me and I'm 100% certain that this could become even **MORE**, certainly **better and efficient**, if approached from the React view point. Something to discuss in our 2pm Thursday meeting.